core/src/test/scala/org/apache/spark/sql/delta/DeltaWriteConfigsSuite.scala

```diff
@@ -0,0 +1,486 @@
+              case "c_or_r_create" | "c_or_r_replace" =>
+                io.delta.tables.DeltaTable.createOrReplace()
+            }
+
+            tblBuilder.addColumn("foo", StringType)
+            tblBuilder = tblBuilder.property(config_no_prefix, "3")
+            tblBuilder = tblBuilder.property(config_prefix, "true")
+
+            val log = (outputLoc, outputMode) match {
+              case ("path", "c_or_r_replace") =>
+                intercept[DeltaAnalysisException] {
+                  tblBuilder.location(dir.getCanonicalPath).execute()
+                }
+                null
+              case ("path", _) =>
+                tblBuilder.location(dir.getCanonicalPath).execute()
+                DeltaLog.forTable(spark, dir)
+              case ("table", _) =>
+                tblBuilder.tableName("tbl").execute()
+                DeltaLog.forTable(spark, TableIdentifier("tbl"))
+            }
+
+            log match {
+              case null =>
+                // CREATE OR REPLACE seems broken when using path and the table already exists
+                // with a different schema.
+                // DeltaAnalysisException: The specified schema does not match the existing schema
+                // ...
+                // Specified schema is missing field(s): bar
+                // Specified schema has additional field(s): foo
+                assert(outputLoc == "path" && outputMode == "c_or_r_replace")
+                dtb_output += ((outputLoc, outputMode, false, false, true))
+              case _ =>
+                val config = log.snapshot.metadata.configuration
+                val answer_no_prefix = config.contains(config_no_prefix)
+                val answer_prefix = config.contains(config_prefix)
+
+                assert(!answer_no_prefix)
+                assert(answer_prefix)
+
+                dtb_output += ((outputLoc, outputMode, answer_no_prefix, answer_prefix, false))
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // scalastyle:off line.size.limit
+  /*
+  SQL Test Output
+  +---------------+-------------------------+--------------+---------+-------------------------+----------------------+--------------------------------+-----------------------------+
+  |Output Location|Config Input             |SQL Operation |AS SELECT|Contains OPTION no-prefix|Contains OPTION prefix|Contains TBLPROPERTIES no-prefix|Contains TBLPROPERTIES prefix|
+  +---------------+-------------------------+--------------+---------+-------------------------+----------------------+--------------------------------+-----------------------------+
+  |path           |options                  |create        |true     |false                    |true                  |false                           |false                        |
+  |path           |options                  |create        |false    |true                     |true                  |false                           |false                        |
```

Comment:
This is the issue mentioned in #809 Right?

Reply:
yes